application/src/main/java/com/sanction/lightning/facebook/FacebookService.java

```diff
@@ -102,17 +104,30 @@ public FacebookUser getFacebookUser() {
    *
    * @return A boolean to denote that the function worked.
    */
-  public FacebookPhoto publishToFacebook(byte[] inputBytes, String fileName, String message) {
-    FacebookPhoto response;
+  public String publishToFacebook(byte[] inputBytes, String type, String fileName,
+                                  String message, String videoTitle) {
+    JsonObject response;
+    List<Parameter> parameters = Lists.newArrayList();
+
+    if (type.equals("photo")) {
+      type = "photos";
+      parameters.add(Parameter.with("message", message));
+    } else if (type.equals("video")) {
+      type = "videos";
+      parameters.add(Parameter.with("description", message));
+      parameters.add(Parameter.with("title", videoTitle));
+    } else {
+      return null;
+    }
 
     try {
-      response = client.publish("me/photos", FacebookPhoto.class,
-              BinaryAttachment.with(fileName, inputBytes), Parameter.with("message", message));
+      response = client.publish("me/" + type, JsonObject.class,
+              BinaryAttachment.with(fileName, inputBytes),
+              parameters.toArray(new Parameter[parameters.size()]));
     } catch (FacebookException e) {
       return null;
     }
-
-    return response;
+    return response.toString();
```

Comment:
I'm pretty sure it just gives the object back in Json format. That's what it's been doing. There may be a better way of doing this, I can look into that in a future PR.